src/OpenLoco/Paint/PaintVehicle.cpp

```diff
@@ -0,0 +1,198 @@
+#include "PaintVehicle.h"
+#include "../CompanyManager.h"
+#include "../Config.h"
+#include "../Graphics/Colour.h"
+#include "../Objects/ObjectManager.h"
+#include "../Objects/VehicleObject.h"
+#include "../Things/Vehicle.h"
+#include "Paint.h"
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::Things::Vehicle;
+
+namespace OpenLoco::Paint
+{
+    // 0x00500160
+    const uint8_t _reversePitch[16]{
+        0, 5, 6, 7, 8, 1, 2, 3, 4, 10, 9, 12, 11, 0, 0, 0
+    };
+
+    // 0x004FFAE8
+    const uint32_t _ghostInvisImageFlags[2]{
+        Gfx::recolour(0, PaletteIndex::index_2C),
+        Gfx::recolourTranslucent(0, PaletteIndex::index_31),
+    };
+
+    // 0x004B0CFC
+    static void paintBogie(PaintSession& session, vehicle_bogie* bogie)
+    {
+        auto* vehObject = ObjectManager::get<vehicle_object>(bogie->object_id);
+        if (bogie->object_sprite_type == 0xFF)
+        {
+            return;
+        }
+
+        auto& sprite = vehObject->bogie_sprites[bogie->object_sprite_type];
+        uint8_t yaw = (bogie->sprite_yaw + (session.getRotation() << 4)) & 0x3F;
```

Comment:
I think we should consider adding an accessor function for this.

Reply:
It's not really an accessor. It's adding to the rotation and rotation has 0x3f bits to it.

Reply:
Would it make sense to have a function like:
```
constexpr uint8_t getViewYaw(const uint8_t spriteYaw, const uint8_t viewRotation)
{
    return (spriteYaw + (viewRotation << 4)) & 0x3F;
}
```

Reply:
Hmm, you're right, it's not doing what I thought it was. Abstracting it into a function doesn't make it more elegant than it is currently.